perception/euclidean_cluster/lib/voxel_grid_based_euclidean_cluster.cpp

```diff
@@ -40,14 +40,27 @@ VoxelGridBasedEuclideanCluster::VoxelGridBasedEuclideanCluster(
   min_points_number_per_voxel_(min_points_number_per_voxel)
 {
 }
-
+// TODO(badai-nguyen): remove this function when field copying also implemented for
+// euclidean_cluster.cpp
 bool VoxelGridBasedEuclideanCluster::cluster(
   const pcl::PointCloud<pcl::PointXYZ>::ConstPtr & pointcloud,
   std::vector<pcl::PointCloud<pcl::PointXYZ>> & clusters)
+{
+  (void)pointcloud;
+  (void)clusters;
+  return false;
+}
+
+bool VoxelGridBasedEuclideanCluster::cluster(
+  const sensor_msgs::msg::PointCloud2::ConstSharedPtr & pointcloud_msg,
+  tier4_perception_msgs::msg::DetectedObjectsWithFeature & objects)
 {
   // TODO(Saito) implement use_height is false version
 
   // create voxel
+  pcl::PointCloud<pcl::PointXYZ>::Ptr pointcloud(new pcl::PointCloud<pcl::PointXYZ>);
+  int point_step = pointcloud_msg->point_step;
+  pcl::fromROSMsg(*pointcloud_msg, *pointcloud);
   pcl::PointCloud<pcl::PointXYZ>::Ptr voxel_map_ptr(new pcl::PointCloud<pcl::PointXYZ>);
```

Comment:
[minor comment]
For readability, I like following style.

```suggestion
  constexpr float z_leaf_size = 100000.0;
  voxel_grid_.setLeafSize(voxel_leaf_size_, voxel_leaf_size_, z_leaf_size);
```